apis/bigquery/v1beta1/zz_generated.resolvers.go

```diff
@@ -0,0 +1,859 @@
+		Selector:     mg.Spec.ForProvider.TableIDSelector,
+		To: reference.To{
+			List:    &TableList{},
+			Managed: &Table{},
+		},
+	})
+	if err != nil {
+		return errors.Wrap(err, "mg.Spec.ForProvider.TableID")
+	}
+	mg.Spec.ForProvider.TableID = reference.ToPtrValue(rsp.ResolvedValue)
+	mg.Spec.ForProvider.TableIDRef = rsp.ResolvedReference
+
+	return nil
+}
+
+// ResolveReferences of this TableIAMMember.
+func (mg *TableIAMMember) ResolveReferences(ctx context.Context, c client.Reader) error {
+	r := reference.NewAPIResolver(c, mg)
+
+	var rsp reference.ResolutionResponse
+	var err error
+
+	rsp, err = r.Resolve(ctx, reference.ResolutionRequest{
+		CurrentValue: reference.FromPtrValue(mg.Spec.ForProvider.DatasetID),
+		Extract:      reference.ExternalName(),
+		Reference:    mg.Spec.ForProvider.DatasetIDRef,
+		Selector:     mg.Spec.ForProvider.DatasetIDSelector,
+		To: reference.To{
+			List:    &DatasetList{},
+			Managed: &Dataset{},
+		},
+	})
+	if err != nil {
+		return errors.Wrap(err, "mg.Spec.ForProvider.DatasetID")
+	}
+	mg.Spec.ForProvider.DatasetID = reference.ToPtrValue(rsp.ResolvedValue)
+	mg.Spec.ForProvider.DatasetIDRef = rsp.ResolvedReference
+
+	rsp, err = r.Resolve(ctx, reference.ResolutionRequest{
+		CurrentValue: reference.FromPtrValue(mg.Spec.ForProvider.TableID),
+		Extract:      reference.ExternalName(),
+		Reference:    mg.Spec.ForProvider.TableIDRef,
+		Selector:     mg.Spec.ForProvider.TableIDSelector,
+		To: reference.To{
+			List:    &TableList{},
+			Managed: &Table{},
+		},
+	})
+	if err != nil {
+		return errors.Wrap(err, "mg.Spec.ForProvider.TableID")
+	}
+	mg.Spec.ForProvider.TableID = reference.ToPtrValue(rsp.ResolvedValue)
+	mg.Spec.ForProvider.TableIDRef = rsp.ResolvedReference
+
+	return nil
+}
+
+// ResolveReferences of this TableIAMPolicy.
+func (mg *TableIAMPolicy) ResolveReferences(ctx context.Context, c client.Reader) error {
+	r := reference.NewAPIResolver(c, mg)
+
+	var rsp reference.ResolutionResponse
+	var err error
+
+	rsp, err = r.Resolve(ctx, reference.ResolutionRequest{
+		CurrentValue: reference.FromPtrValue(mg.Spec.ForProvider.DatasetID),
+		Extract:      resource.ExtractParamPath("dataset_id", false),
+		Reference:    mg.Spec.ForProvider.DatasetIDRef,
+		Selector:     mg.Spec.ForProvider.DatasetIDSelector,
+		To: reference.To{
+			List:    &TableList{},
+			Managed: &Table{},
+		},
+	})
+	if err != nil {
+		return errors.Wrap(err, "mg.Spec.ForProvider.DatasetID")
+	}
+	mg.Spec.ForProvider.DatasetID = reference.ToPtrValue(rsp.ResolvedValue)
+	mg.Spec.ForProvider.DatasetIDRef = rsp.ResolvedReference
+
+	rsp, err = r.Resolve(ctx, reference.ResolutionRequest{
+		CurrentValue: reference.FromPtrValue(mg.Spec.ForProvider.Project),
+		Extract:      resource.ExtractParamPath("project", false),
+		Reference:    mg.Spec.ForProvider.ProjectRef,
+		Selector:     mg.Spec.ForProvider.ProjectSelector,
+		To: reference.To{
+			List:    &TableList{},
+			Managed: &Table{},
+		},
+	})
```

Comment:
is this a valid reference?

Reply:
Thanks for catching. Again `project_id` [refers to](https://registry.terraform.io/providers/hashicorp/google/4.22.0/docs/resources/bigquery_table_iam#google_bigquery_table_iam_policy) a table's `project_id`.